src/common/snippets/src/lowered/pass/pass.cpp

```diff
@@ -16,23 +16,33 @@ PassPipeline::PassPipeline(const std::shared_ptr<PassConfig>& pass_config) : m_p
     OPENVINO_ASSERT(m_pass_config != nullptr, "PassConfig is not initialized!");
 }
 
-void PassPipeline::register_pass(const snippets::pass::PassPosition& position, const std::shared_ptr<Pass>& pass) {
+void PassPipeline::register_pass(const snippets::pass::PassPosition& position, const std::shared_ptr<PassBase>& pass) {
     OPENVINO_ASSERT(pass != nullptr, "PassPipeline cannot register empty pass!");
     m_passes.insert(position.get_insert_position(m_passes), pass);
 }
 
-void PassPipeline::register_pass(const std::shared_ptr<Pass>& pass) {
+void PassPipeline::register_pass(const std::shared_ptr<PassBase>& pass) {
     OPENVINO_ASSERT(pass != nullptr, "PassPipeline cannot register empty pass!");
     m_passes.push_back(pass);
 }
 
 void PassPipeline::run(LinearIR& linear_ir) const {
+    run(linear_ir, linear_ir.begin(), linear_ir.end());
+}
+
+void PassPipeline::run(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end) const {
```

Comment:
BTW can we check that there iterators from this `LinearIR`? To avoid comparison of iterators from different collections

Reply:
I added a lambda which checks that and found an issue. Some time on investigation is needed

Reply:
The only method to check that some target iterator belong to a std::list container is to iterate through the whole container and compare each iterator with the target iterator. This is quite expensive check which should be performed each time when handlers are called => this happens many times during `InsertSpecificIterations` pass execution.
Basing on that, it was agreed that the check is too expensive to add it in product code.